lib/cocoapods/installer/xcode/pods_project_generator/target_installation_result.rb

```diff
@@ -0,0 +1,88 @@
+          #         The target this installation result is for.
+          #
+          attr_reader :target
+
+          # @return [PBXNativeTarget] native_target
+          #         The native target that was produced for this target.
+          #
+          attr_reader :native_target
+
+          # @return [Array<PBXNativeTarget>] resource_bundle_targets
+          #         The resource bundle targets that were produced for this target. Can be empty if the target had
+          #         no resource bundles.
+          #
+          attr_reader :resource_bundle_targets
+
+          # @return [Array<PBXNativeTarget>] test_native_targets
+          #         The test native targets that were produced for this target. Can be empty if there were no test
+          #         native targets created (e.g. no test specs present).
+          #
+          attr_reader :test_native_targets
+
+          # @return [Array<PBXNativeTarget>] test_resource_bundle_targets
+          #         The test resource bundle targets that were produced for this target. Can be empty if the target had
+          #         no resource bundles for any tests.
+          #
+          attr_reader :test_resource_bundle_targets
+
+          # @return [Array<PBXNativeTarget>] test_app_host_targets
+          #         The test app host native targets that were produced for this target. Can be empty.
+          #
+          attr_reader :test_app_host_targets
+
+          # Initialize a new instance
+          #
+          # @param [Target] target @see #target
+          # @param [PBXNativeTarget] native_target @see #native_target
+          # @param [Array<PBXNativeTarget>] resource_bundle_targets @see #resource_bundle_targets
+          # @param [Array<PBXNativeTarget>] test_native_targets @see #test_native_targets
+          # @param [Array<PBXNativeTarget>] test_resource_bundle_targets @see #test_resource_bundle_targets
+          # @param [Array<PBXNativeTarget>] test_app_host_targets @see #test_app_host_targets
+          #
+          def initialize(target, native_target, resource_bundle_targets = [], test_native_targets = [],
+                         test_resource_bundle_targets = [], test_app_host_targets = [])
+            @target = target
+            @native_target = native_target
+            @resource_bundle_targets = resource_bundle_targets
+            @test_native_targets = test_native_targets
+            @test_resource_bundle_targets = test_resource_bundle_targets
+            @test_app_host_targets = test_app_host_targets
+          end
+
+          # Returns the corresponding native target to use based on the provided specification.
+          #
+          # @param  [Specification] spec
+          #         The specification to base from in order to find the native target.
+          #
+          # @return [PBXNativeTarget] the native target to use or `nil` if none is found.
+          #
+          def native_target_for_spec(spec)
```

Comment:
I think this logic is duplicated above, we should use this method instead

Reply:
done